cluster-autoscaler/core/scaleup/orchestrator/orchestrator.go

```diff
@@ -362,6 +362,18 @@ func (o *ScaleUpOrchestrator) ScaleUpToNodeGroupMinSize(
 		return scaleUpError(&status.ScaleUpStatus{}, aErr.AddPrefix("could not compute total resources: "))
 	}
 
+	nodeCountInCluster := 0
```

Comment:
I think this variable should also contain nodes that are upcoming, i.e. are being created by the cloud provider (and didn't yet register in K8s).
The standard ScaleUp function uses a following call to fetch this information: 
```
upcomingCounts, _ := o.clusterStateRegistry.GetUpcomingNodes()
```
upcomingCounts contains a map from the nodegroup name to a number of VM so something as follows should add this information:
```
for _, numberOfNodes := range upcomingCounts {
	nodeCountInCluster += numberOfNodes
}
```

Reply:
I believe `ng.TargetSize()` already includes the latest node count including upcoming nodes. This is a direct call to cloud provider, and cloud provider will return the real-time desired node count of the node group at that moment. Your point would be valid if I used `len(nodes)` here. Please let me know if you have different ideas.

Reply:
Yup, you are right.